dev-docs/007-transforms-rfc.md

```diff
@@ -0,0 +1,99 @@
+- Start Date: 2019-09-05
+- RFC PR: 
+
+# Summary
+
+It is desirable to enable arbitrary transform support common in robotic systems. While XVIZ models some explicitly
+with our coordinate types IDENTITY, GEOGRAPHIC, and VEHICLE_RELATIVE, the need for client controlled frames of reference
+will make it easier to interop with other systems as well as match a familiar modeling automous system data.
+
+# Motivation
+
+Some PERSISTENT data, such as *maps* and *point clouds*, could be encoded in various ways such as vehicle relative.
+If they are semantically fixed, but the reference point changes, there is no way to efficiently update them without
+sending all the points again.
+
+Ideally, the only data that is required is the transform to adjust how the points are transformed.
+
+Extending this to a general robotic systems convention, the need to describe parent/child relationships
+is a common model for robotic systems.
+
+XVIZ is not focused on articulated robotic platforms but support for the convention common when modeling autonomous data
+would be benefitial for XVIZ and it's users.
+
+# Proposal
+
+### POSE schema
+ - Add a *parent* field to the pose that is a string that my reference another pose stream
+   If present within the data, it would be used to construct the final transform matrix for the data
+ - Poses may reference any other pose as a *parent* across all state_update message types. Meaning a PERSISTENT
+   pose could be a parent of a *COMPLETE* or *INCREMENTAL* pose.
+ - The pose stream names must be unique.
+ - Remove timestamp as required field on the pose schema. The state_update defines the timestamp for the message already
+   and the pose entry was just a carry over from xviz v1
+
+### PRIMITIVE schema
+ - add a *pose* field to the primitive_base schema, making the attribute accessible to all primitives
+
+### STREAM METADATA schema
+ - Add a *pose* field, as a string referencing a POSE stream
```

Comment:
Does this apply to vehicle relative & identity or just dynamic coordinates?

Reply:
I think so.  In fact, IDENTITY and VEHICLE_RELATIVE are where I was thinking most.  I'm not sure how it would work with DYNAMIC yet.

Alternatively, we could add a setting SCENE_GRAPH, which sets the expectation for a stream and keeps it independent.  That may be easier than dealing with the interactions between the current COORDINATES or creating invalid combinations that we have to check for and handle.

Do you have any preference? I'm still thinking through, but SCENE_GRAPH seems appealing (name obviously change)

Reply:
Short answer with the new approach is *yes*